### DISCUSSION TITLE
One Large Polygon with holes Output - Annotations is  fine but masks give all white with some black patches

### DISCUSSION THREAD

**Discussion by ibrahimnzamane** [opengeos/segment-geospatial#326] [Q&A] [answered]
I am running sam2 on a satellite imagery scaled to 256 bit. I am getting good results with unique objects in show_anns() but when I export it using raster_to_vector or show mask I get one continous white with small black patches in raster and large polygon with small holes in vector. I have set unique = true in generate method but still get this error. 

My guess is I am getting foreground/background and a large area is segmented as object. How do I efficiently export unique segmetns in raster or How do I disable the large area (~70% of image) segments from the mask?

**Comment by giswqs** [comment]
Thank you for reporting. I am aware of this issue. I will look into it

**Comment by giswqs** [comment, marked as answer]
This bug has been fixed in https://github.com/opengeos/segment-geospatial/pull/332

**Comment by giswqs** [comment]
Also added support for filtering objects by size. 
https://github.com/opengeos/segment-geospatial/pull/333
